GraphColoring/Tests.qs

```diff
@@ -267,4 +272,103 @@ namespace Quantum.Kata.GraphColoring {
             Message($"Got correct coloring {coloring}");
         }
     }
+
+    @Test("QuantumSimulator")
+    operation T31_DoesEdgeStartWithVertex () : Unit {
+        Fact(DoesEdgeStartWithVertex( (1,2), 1) == true,
+             $"Edge (1,2) judged incorrect for vertex 1");
+        Fact(DoesEdgeStartWithVertex( (1,2), 2) == false,
+             $"Edge (1,2) judged correct for vertex 2");
+        Fact(DoesEdgeStartWithVertex( (1,2), 3) == false,
+             $"Edge (1,2) judged correct for vertex 2");
+    }
+
+    @Test("QuantumSimulator")
+    operation T321_IsVertexWeaklyColored () : Unit {
+        let testCases = ExampleGraphs();
+        let colorings = [[0, 0, 0],
+                         [3, 2, 0, 0],
+                         [1, 0, 1, 2, 1],
+                         [0, 1, 1, 1, 1],
+                         [0, 1, 2, 2, 2]
+                        ];
+        let expectedResults = [[true, true, true],
+                               [true, true, false, true],
+                               [true, true, true, true, false],
+                               [true, false, true, false, false],
+                               [true, true, false, false, true]
+                              ];
+        for t in 0..Length(testCases)-2 {
+            let (V_t, edges_t) = testCases[t];
+            for vertex in 0 .. V_t - 1 {
+            Fact(IsVertexWeaklyColored(V_t, edges_t, colorings[t], vertex) == expectedResults[t][vertex],
+                $"Vertex = {vertex} judged {not expectedResults[t][vertex]} for coloring = {colorings[t]} , edges = {edges_t}");
+            }
+        }
+    }
+
+    @Test("QuantumSimulator")
+    operation T322_IsWeakColoringValid () : Unit {
+        let testCases = ExampleGraphs();
+        let exampleColoringForThreeVertices = [[0, 0, 0], [2, 1, 3]];
```

Comment:
Are there any test cases that are interesting for the weak graph coloring but not for the regular graph coloring, both in terms of graph structure and its coloring? It makes sense to add notes on the way you selected the colorings for the graphs to see if any corner cases are missing, similar to the comments on the way we selected example graphs themselves.

Reply:
Hi added a note for the same. I just explored the current graphs from the Coloring perspective. Using trivial cases to make things simple. The interesting part about these graphs is that since graph 5 is a superset of graph 3 and 4, any weak coloring that passes graph 3 or 4, would also pass graph 5